ibc/utils_test.go

```diff
@@ -370,3 +370,56 @@ func TestDeriveDecimalsFromDenom(t *testing.T) {
 		require.Equal(t, tc.expDec, dec)
 	}
 }
+
+// On the OnRecv function there are two denominations
+// coin.Denom contains IBC denomination (e.g aevmos, IBC/hash)
+// data.Denom contains source chain denomination (e.g transfer/channel-x/aevmos, uosmo)
+
+// In this function we are checking the source chain denomination (2nd case).
+// On osmosis chain coins will look like:
+// uosmo <- less than 2 slashes is native from source chain
+// transfer/channel-xx/atom <- 2 slashes is not native. So its a hop coin.
+func TestIsSingleHop(t *testing.T) {
```

Comment:
It might make sense to even rename this to `IsNativeOnSourceChain` 🤔 but maybe that's also overthinking it haha

Reply:
Agree. Gonna rename the function so its less confusing,